src/backend/executor/spi.c

```diff
@@ -1709,8 +1709,9 @@ SPI_cursor_open_internal(const char *name, SPIPlanPtr plan,
 		if (list_length(stmt_list) == 1 &&
 			linitial_node(PlannedStmt, stmt_list)->commandType != CMD_UTILITY &&
 			linitial_node(PlannedStmt, stmt_list)->rowMarks == NIL &&
+			!linitial_node(PlannedStmt, stmt_list)->parallelModeNeeded &&
```

Comment:
change to 
```
( !linitial_node(PlannedStmt, stmt_list)->parallelModeNeeded && sql_dialect == TSQL_DIALECT )
```

make sure it happen only in babel

Reply:
I just pushed the change for this